homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,145 @@
+            )
+            return await self.async_step_auto()
+
+        if import_config[CONF_TYPE] == "manual":
+            for entry in config_entry:
+                if (
+                    entry.unique_id
+                    == f"{DOMAIN}_{import_config[CONF_HOST].replace('.','_')}"
+                ):
+                    _LOGGER.error(
+                        "Your flux_led configuration for %s has already been imported. Please remove configuration from your configuration.yaml.",
+                        import_config[CONF_HOST],
+                    )
+                    return self.async_abort(reason="already_configured_device")
+
+            _LOGGER.error(
+                "Imported flux_led configuration for %s. Please remove from your configuration.yaml.",
+                import_config[CONF_HOST],
+            )
+            return await self.async_step_manual(import_config, source_import=True)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        config_type = None
+
+        if user_input is not None:
+            config_type = user_input[CONF_TYPE]
+
+            return (
+                await self.async_step_auto()
+                if config_type == "auto"
+                else await self.async_step_manual()
+            )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_TYPE, default="auto"): vol.In(
+                        {
+                            "auto": "Auto Search",
+                            "manual": "Manual Configuration",
+                        }
+                    )
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_auto(self, user_input=None):
+        """Complete the auto configuration step for setup."""
+        bulb_scanner = BulbScanner()
+        devices = bulb_scanner.scan()
+
+        if not devices:
+            return self.async_abort(reason="no_devices_found")
+
+        user_data = {
+            CONF_TYPE: "auto",
+        }
+        unique_id = "flux_led_auto"
+        await self.async_set_unique_id(unique_id)
+
+        return self.async_create_entry(
+            title="Auto Search",
+            data=user_data,
+        )
+
+    async def async_step_manual(self, user_input=None, source_import: bool = False):
+        """Complete the manual setup of an individual FluxLED Light."""
+        errors = {}
+
+        if user_input is not None:
+            name = user_input[CONF_NAME]
+            ip_address = user_input[CONF_HOST]
+
+            try:
+                bulb = WifiLedBulb(ipaddr=ip_address)
+                bulb.update_state()
+
+                if bulb.mode:
+                    user_data = {
+                        CONF_TYPE: "manual",
```

Comment:
I don't think we should mix manual config and auto mode in the config entry data.

We should create a single config entry for the integration and use options flow to set discovery on/off and configure specific devices.

The default should be to use discovery.

Reply:
So in the case that discovery finds no devices the integration should still be installed rather than aborting? I took the approach of adding a single instance of the integration for each of either 1 x auto configuration, 1 per manual configuration device to allow flexibility by device.

If I use options for manual it felt more complicated to support and manage multiple manually configured devices.

Reply:
Yeah, in the case where no devices are discovered, the integration will be setup but no entities will have been added yet. There should probably be some periodic discovery so that devices can be discovered when they become available and then the corresponding entities will be added.

Reply:
Additional reason to run separate instances would be that I plan to add an options flow by device to allow default effect speed to be set for each light separately.

Reply:
If we're not doing period discovery now, we can add that later.

Reply:
I have periodic discovery included in the dataupdatecoordinator PR for auto configured instance.

Thoughts on the device specific options/configuration? Is there another integration that works this way (discovery and then manual addition of multiple devices via options flow) that I can look at?

Reply:
The rfxtrx integration has something similar with an options flow with both global options and device specific options.

Reply:
OK - interesting. I'm not sure the benefit of isolating to the options flow vs. managing 1 device per instance for manually configured devices. The configuration of the rfxtrx seems more complicated from a UI flow for setup.

Is there a resource overhead difference when we load multiple instances of the same integration? Just want to make sure I understand the problem we are trying to solve by taking one approach vs. another.

Reply:
We prefer to have one config entry for the integration over one entry per device. It's also weird to mix the config entry data structure, where some entries would have connection details and others not. The config entry data structure should be the same for all entries for an integration.

Reply:
OK - will do some playing and see what I can structure that way.

Reply:
Addressed and restructured in the latest commit. Once the config_flow is ok I can rework the update coordinators etc in the other related PRs to match.